### PR TITLE
Fixed flatpak module to accept a list of names to install/uninstall

### PR DESCRIPTION
**Pull request by jdufresne**
##### SUMMARY

Fixes ansible/ansible#41996

To match consistency with other packaging modules, flatpak now accepts a
list packages to install or uninstall. This behavior is described in 2.7
porting guide under the heading "Using a loop on a package module via
squash_actions".

https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.7.html#using-a-loop-on-a-package-module-via-squash-actions

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
flatpak

### CODE REVIEW

**Comment by jdufresne** [comment]
> Most notably, it treats commands to be executed as strings (they should be lists so that there are no escaping issues!

Done in #269 :slightly_smiling_face:

File: changelogs/fragments/267-flatpak-install-list.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- flatpak - allow installing a list of packages.
```
Comment (Andersson007) [inline changelogs/fragments/267-flatpak-install-list.yaml:2]:
```suggestion
- flatpak - allow installing a list of packages (https://github.com/ansible/ansible/issues/41996).
```

**Comment by jdufresne** [comment]
Thanks for the review. I added two new examples. One for installing multiple packages and one for removing.

**Comment by felixfontein** [comment]
@jdufresne could you please rebase your branch? And try to add a test which installs two packages at the same time if possible?

**Comment by jdufresne** [comment]
I have rebased, thanks.

> And try to add a test which installs two packages at the same time if possible?

I tried, but I was unsuccessful getting the test environment up and running. I wasn't even able to run the existing test suite, so I'm not confident in creating a new one. If there are guides on getting this up and running, I could give it another go.

If anyone else is able to contribute the test, that would be helpful too.

I did run manual tests through a number of scenarios with my local scripts.

**Comment by felixfontein** [comment]
Have you looked at https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html#testing-collections ?

**Comment by Andersson007** [comment]
@jdufresne  use the link provided by @felixfontein to run tests in docker containers. Anyway, you can add tests to existing tests/integration/targets/flatpak, push them to here and they will run the same way in RH testing cloud. You'll see in the shippable console if something goes wrong.

**Comment by jdufresne** [comment]
>  Anyway, you can add tests to existing tests/integration/targets/flatpak, push them to here and they will run the same way in RH testing cloud. You'll see in the shippable console if something goes wrong.

Thanks for this suggestion, I took this approach and added several tests. Let me know if you'd like to see anything else included. Cheers.

**Comment by felixfontein** [comment]
@jdufresne did the tests run though for you? They are not running in CI unfortunately (https://github.com/ansible-collections/community.general/blob/13d03a6ce459398dc21670b264a06c0a1200e501/tests/integration/targets/flatpak/aliases#L1).

**Comment by Andersson007** [comment]
@jdufresne any updates on this?

**Comment by gundalow** [comment]
bot_status

**Comment by gundalow** [comment]
@jdufresne Could you answer the questions above about testing.
Also this PR needs a rebase

Please shout out if you need any help

**Comment by jdufresne** [comment]
Thanks for coming back to check on this.

I am no longer working on this and will not have the time to continue. If someone would like to pick it up, complete the abobve, and take it over the finish line, they are welcome to. Thanks.

**Comment by bmillemathias** [comment]
I ran the integration test successfully
```
17:36 $ sudo /home/baptistemm/Code/venv/ansible-stable/bin/ansible-test integration flatpak --allow-unsupported --allow-destructive --allow-root --docker --docker-privileged                                                                                                   
Running flatpak integration test role                                                                                                   
[WARNING]: running playbook inside collection community.general                                                                         
                                                                    
PLAY [testhost] ****************************************************************                                                        
                                                                                                                                        
TASK [Gathering Facts] *********************************************************                                                        
ok: [testhost]                                                                                                                          
                                                                                                                                        
TASK [Install flatpak on Fedora] ***********************************************                                                        
skipping: [testhost]                                                                                                                    
                                                                                                                                        
TASK [Activate flatpak ppa on Ubuntu] ******************************************                                                        
changed: [testhost]                                                                                                                     
                                                                                                                                        
TASK [Install flatpak package on Ubuntu] ***************************************                                                        
changed: [testhost]                                                                                                                     
                                                                                                                                        
TASK [flatpak : Enable flathub for user] ***************************************                                                        
changed: [testhost]                                                                                                                     
                                                                                                                                        
TASK [flatpak : Enable flathub for system] *************************************                                                        
changed: [testhost]                                                                                                                     
                                                                                                                                        
TASK [Add flatpak for testing check mode on present flatpak] *******************                                                        
changed: [testhost]                                                                                                                     
                                                                                                                                        
TASK [flatpak : Test executable override] **************************************                                                        
fatal: [testhost]: FAILED! => {"changed": false, "msg": "Executable 'nothing-that-exists' was not found on the system."}                
...ignoring                                                                                                                             
                                                                    
TASK [flatpak : Verify executable override test result] ************************                                                        
ok: [testhost] => {                                                                                                                     
    "changed": false,                                               
    "msg": "All assertions passed"                                                                                                      
}                                                                                                                                       
                                                                                                                                        
TASK [Test addition of absent flatpak (check mode)] ****************************                                                                                                                                                                                      [343/1985]
changed: [testhost]                                                                                                                     
                                                                                                                                                                                                                                                                                
TASK [Verify addition of absent flatpak test result (check mode)] **************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test non-existent idempotency of addition of absent flatpak (check mode)] ***
changed: [testhost]

TASK [Verify non-existent idempotency of addition of absent flatpak test result (check mode)] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test removal of absent flatpak check mode] *******************************
ok: [testhost]

TASK [Verify removal of absent flatpak test result (check mode)] ***************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test addition of absent flatpak with url (check mode)] *******************
changed: [testhost]

TASK [Verify addition of absent flatpak with url test result (check mode)] *****
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test non-existent idempotency of addition of absent flatpak with url (check mode)] ***
changed: [testhost]

TASK [Verify non-existent idempotency of additionof absent flatpak with url test result (check mode)] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test removal of absent flatpak with url not doing anything (check mode)] ***
ok: [testhost]

TASK [Verify removal of absent flatpak with url test result (check mode)] ******
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test addition of present flatpak (check mode)] ***************************
ok: [testhost]

TASK [Verify addition test result of present flatpak (check mode)] *************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test removal of present flatpak (check mode)] ****************************
changed: [testhost]

TASK [Verify removal of present flatpak test result (check mode)] **************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test non-existent idempotency of removal (check mode)] *********
changed: [testhost]

TASK [flatpak : Verify non-existent idempotency of removal (check mode)] *******
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test addition with url of present flatpak (check mode)] ******************
ok: [testhost]

TASK [Verify addition with url of present flatpak test result (check mode)] ****
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test removal with url of present flatpak (check mode)] *******************
changed: [testhost]

TASK [Verify removal with url of present flatpak test result (check mode)] *****
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [Test non-existent idempotency of removal with url of present flatpak (check mode)] ***
changed: [testhost]

TASK [Verify non-existent idempotency of installation with url of present flatpak test result (check mode)] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}


TASK [flatpak : Test addition - user] ******************************************
changed: [testhost]
                                                                    
TASK [flatpak : Verify addition test result - user] ****************************                                                        
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of addition - user] ***************************
ok: [testhost]

TASK [flatpak : Verify idempotency of addition test result - user] *************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test removal - user] *******************************************
changed: [testhost]

TASK [flatpak : Verify removal test result - user] *****************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of removal - user] ****************************
ok: [testhost]

TASK [flatpak : Verify idempotency of removal test result - user] **************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test addition with url - user] *********************************
changed: [testhost]

TASK [flatpak : Verify addition test result - user] ****************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of addition with url - user] ******************
ok: [testhost]

TASK [flatpak : Verify idempotency of addition with url test result - user] ****
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}


TASK [flatpak : Test removal with url - user] **********************************
changed: [testhost]

TASK [flatpak : Verify removal test result - user] *****************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of removal with url - user] *******************
ok: [testhost]

TASK [flatpak : Verify idempotency of removal with url test result - user] *****
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test addition with list - user] ********************************
changed: [testhost]

TASK [flatpak : Verify addition with list test result - user] ******************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of addition with list - user] *****************
ok: [testhost]

TASK [flatpak : Verify idempotency of addition with list test result - user] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test addition with list partially installed - user] ************
changed: [testhost]

TASK [flatpak : Verify addition with list partially installed test result - user] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of addition with list partially installed - user] ***
ok: [testhost]

TASK [flatpak : Verify idempotency of addition with list partially installed test result - user] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}


TASK [flatpak : Test removal with list - user] *********************************
changed: [testhost]

TASK [flatpak : Verify removal with list test result - user] *******************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of removal with list - user] ******************
ok: [testhost]

TASK [flatpak : Verify idempotency of removal with list test result - user] ****
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test removal with list partially removed - user] ***************
changed: [testhost]

TASK [flatpak : Verify removal with list partially removed test result - user] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of removal with list partially removed - user] ***
ok: [testhost]

TASK [flatpak : Verify idempotency of removal with list partially removed test result - user] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test addition - system] ****************************************

changed: [testhost]

TASK [flatpak : Verify addition test result - system] **************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of addition - system] *************************
ok: [testhost]

TASK [flatpak : Verify idempotency of addition test result - system] ***********
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}


TASK [flatpak : Test removal - system] *****************************************
changed: [testhost]

TASK [flatpak : Verify removal test result - system] ***************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of removal - system] **************************
ok: [testhost]

TASK [flatpak : Verify idempotency of removal test result - system] ************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test addition with url - system] *******************************
changed: [testhost]

TASK [flatpak : Verify addition test result - system] **************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of addition with url - system] ****************
ok: [testhost]

TASK [flatpak : Verify idempotency of addition with url test result - system] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test removal with url - system] ********************************
changed: [testhost]

TASK [flatpak : Verify removal test result - system] ***************************
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [flatpak : Test idempotency of removal with url - system] *****************
ok: [testhost]

TASK [flatpak : Verify idempotency of removal with url test result - system] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY RECAP *********************************************************************
testhost                   : ok=96   changed=29   unreachable=0    failed=0    skipped=1    rescued=0    ignored=1   
```

**Comment by russoz** [comment]
Per the author's comment stating he is no longer working on this, per the fact that the branch is no longer in github, and per the fact this PR now has conflicts with the main codebase, I transplanted the changes to a new PR (#2521) and I am closing this one.

**Pull request closed without merging**